src/main/java/rx/internal/operators/OperatorConcat.java

```diff
@@ -112,9 +112,19 @@ public void onStart() {
         }
 
         private void requestFromChild(long n) {
-            if (n <=0) return;
+            if (n <= 0) return;
             // we track 'requested' so we know whether we should subscribe the next or not
-            long previous = BackpressureUtils.getAndAddRequest(requested, n);
+            
+            final AtomicLong requestedField = requested;
+            
+            long previous;
+            
+            if (requestedField.get() != Long.MAX_VALUE) {
+                previous = BackpressureUtils.getAndAddRequest(requestedField, n);
```

Comment:
No, the `getAndAddRequest` doesn't check for max-value and always does the CAS. I'd leave this as is and have a separate PR where the `getAndAddRequest` is changed and all use places are verified/fixed.